donkeycar/parts/mavlink.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python3
+from __future__ import print_function
+"""
+Script for using a Pixhawk running any ArduPilot- or PX4-based code with Donkeycar 
+
+author: @zlite (Chris Anderson), 2022
+contrib: @wallarug (Cian Byrne) 2019
+contrib: @peterpanstechland 2019
+contrib: @sctse999 2020
+
+If you're getting a permissions error on the USB port: "sudo chmod 666 /dev/ttyACM0"
+
+requires: 
+--gcc (sudo apt install gcc)
+--pymavlink (pip install pymavlink)
+--docopt (pip install docopt)
+
+"""
+
+import time
+import donkeycar as dk
+import config as cfg
+
+debug = False
+
+try:
+    import serial
+except ImportError:
+    print("PySerial not found.  Please install: pip install pyserial")
+
+try:
+    from pymavlink import mavutil
+except ImportError:
+    print("Pymavlink not found.  Please install: pip install pymavlink")
+
+
+class MavlinkController:
+    '''
+    Driver to read MAVLink RC data from a Pixhawk, for manual control
+    '''
+
+    def __init__(self, cfg, debug=False):
```

Comment:
Logging should be set on a global level and then using the `logging` module like in other parts.